libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp

```diff
@@ -87,7 +87,7 @@ namespace hpx::execution::experimental::detail {
         // sync_wait or sync_wait_with_variant by checking Sync_wait_type is single or variant
         using result_type = std::conditional<Type == Sync_wait_type::single,
             hpx::variant<single_result_type>,
-            predecessor_value_types<hpx::tuple, hpx::variant>>::type;
+            predecessor_value_types<hpx::tuple, hpx::variant>>; // ::type
 
```

Comment:
`
Consolidate compiler generated dependencies of target hpx_parcelset_base
[ 30%] Building CXX object libs/full/parcelset_base/CMakeFiles/hpx_parcelset_base.dir/src/parcelport.cpp.o
In file included from /home/chuanqiu/project1/new_hpx/src/hpx/buildtest/libs/core/execution/include/hpx/modules/execution.hpp:26,
                 from /home/chuanqiu/project1/new_hpx/src/hpx/libs/core/runtime_local/include/hpx/runtime_local/service_executors.hpp:13,
                 from /home/chuanqiu/project1/new_hpx/src/hpx/libs/core/runtime_local/include/hpx/runtime_local/run_as_os_thread.hpp:14,
                 from /home/chuanqiu/project1/new_hpx/src/hpx/buildtest/libs/core/runtime_local/include/hpx/modules/runtime_local.hpp:29,
                 from /home/chuanqiu/project1/new_hpx/src/hpx/libs/full/parcelset_base/src/parcelport.cpp:16:
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:86:13: error: template argument 1 is invalid
   86 |             single_variant_t<predecessor_value_types<hpx::tuple, meta::pack>>;
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:90:26: error: ‘single_result_type’ was not declared in this scope; did you mean ‘single_result_t’?
   90 |             hpx::variant<single_result_type>,
      |                          ^~~~~~~~~~~~~~~~~~
      |                          single_result_t
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:90:44: error: template argument 1 is invalid
   90 |             hpx::variant<single_result_type>,
      |                                            ^
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:91:61: error: template argument 2 is invalid
   91 |             predecessor_value_types<hpx::tuple, hpx::variant>>::type;
      |                                                             ^~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:109:54: error: ‘result_type’ was not declared in this scope
  109 |             hpx::variant<hpx::monostate, error_type, result_type> value;
      |                                                      ^~~~~~~~~~~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:109:65: error: template argument 3 is invalid
  109 |             hpx::variant<hpx::monostate, error_type, result_type> value;
      |                                                                 ^
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp: In member function ‘auto hpx::execution::experimental::detail::sync_wait_receiver<Sender, Type>::shared_state::get_value()’:
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:127:42: error: ‘single_result_type’ was not declared in this scope; did you mean ‘single_result_t’?
  127 |                     return hpx::optional<single_result_type>(
      |                                          ^~~~~~~~~~~~~~~~~~
      |                                          single_result_t
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:127:60: error: template argument 1 is invalid
  127 |                     return hpx::optional<single_result_type>(
      |                                                            ^
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:128:46: error: ‘result_type’ was not declared in this scope
  128 |                         hpx::get<0>(hpx::get<result_type>(HPX_MOVE(value))));
      |                                              ^~~~~~~~~~~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:133:34: error: ‘result_type’ was not declared in this scope
  133 |                         hpx::get<result_type>(HPX_MOVE(value)));
      |                                  ^~~~~~~~~~~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:144:38: error: ‘result_type’ was not declared in this scope
  144 |                 return hpx::optional<result_type>();
      |                                      ^~~~~~~~~~~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:144:49: error: template argument 1 is invalid
  144 |                 return hpx::optional<result_type>();
      |                                                 ^
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp: In function ‘void hpx::execution::experimental::detail::tag_invoke(hpx::execution::experimental::set_value_t, hpx::execution::experimental::detail::sync_wait_receiver<Sender, Type>&&, Us&& ...)’:
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:179:44: error: ‘result_type’ was not declared in this scope
  179 |             r.state.value.template emplace<result_type>(
      |                                            ^~~~~~~~~~~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp: At global scope:
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:221:43: error: ‘Type’ has not been declared
  221 |             sync_wait_t, Sender&& sender, Type&& tp)
      |                                           ^~~~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp: In function ‘constexpr auto hpx::this_thread::experimental::tag_fallback_invoke(hpx::this_thread::experimental::sync_wait_t, Sender&&, int&&)’:
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:224:17: error: expected type-specifier
  224 |                 hpx::execution::experimental::detail::<Sender, Type>;
      |                 ^~~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:225:41: error: ‘receiver_type’ has not been declared
  225 |             using state_type = typename receiver_type::shared_state;
      |                                         ^~~~~~~~~~~~~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:227:13: error: ‘state_type’ was not declared in this scope; did you mean ‘state_stopped’?
  227 |             state_type state{};
      |             ^~~~~~~~~~
      |             state_stopped
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:229:46: error: ‘receiver_type’ was not declared in this scope
  229 |                 HPX_FORWARD(Sender, sender), receiver_type{state});
      |                                              ^~~~~~~~~~~~~
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:232:18: error: expected unqualified-id before ‘.’ token
  232 |             state.wait();
      |                  ^
/home/chuanqiu/project1/new_hpx/src/hpx/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:233:25: error: expected primary-expression before ‘.’ token
  233 |             return state.get_value();
      |                         ^
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-interference-size’
make[2]: *** [libs/full/parcelset_base/CMakeFiles/hpx_parcelset_base.dir/build.make:132: libs/full/parcelset_base/CMakeFiles/hpx_parcelset_base.dir/src/parcelport.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:79628: libs/full/parcelset_base/CMakeFiles/hpx_parcelset_base.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
`